tests/marker_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests_test
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kubevirt.io/ovs-cni/tests"
+)
+
+var _ = Describe("ovs-cni-marker", func() {
+	Describe("bridge resource reporting", func() {
+
+		Context("when bridge is configured on a node", func() {
+			out, err := tests.RunOnNode("node01", "sudo ovs-vsctl add-br br-test")
+			if err != nil {
+				panic(fmt.Errorf("%v: %s", err, out))
+			}
+			defer tests.RunOnNode("node01", "sudo ovs-vsctl --if-exists del-br br-test")
+
+			It("should be reported as a node resource", func() {
+				Eventually(func() bool {
+					node, err := clientset.CoreV1().Nodes().Get("node01", v1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					capacity, reported := node.Status.Capacity["ovs-cni.network.kubevirt.io/br-test"]
+					if !reported {
+						return false
+					}
+					capacityInt, _ := capacity.AsInt64()
+					if capacityInt != int64(1000) {
+						return false
+					}
+					return true
+				}, 20*time.Second, 5*time.Second).Should(Equal(true))
+			})
+		})
+
+		Context("when bridge is not available anymore", func() {
+			out, err := tests.RunOnNode("node01", "sudo ovs-vsctl --if-exists del-br br-test")
+			if err != nil {
+				panic(fmt.Errorf("%v: %s", err, out))
+			}
+
+			It("should not be reported as a node resource", func() {
```

Comment:
shouldn't you first check that it WAS reported, to make sure that you validate real cleanup and not just failure to update capacity in any way?

Reply:
The previous context covers that, doesn't it? What should be different?